codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -2865,32 +2866,58 @@ loadProjectBranchSql :: Sql
 loadProjectBranchSql =
   [sql|
     SELECT
-      project_id,
-      branch_id,
-      name
+      project_branch.project_id,
+      project_branch.branch_id,
+      project_branch.name,
+      project_branch_parent.parent_branch_id
     FROM
       project_branch
+      LEFT JOIN project_branch_parent ON project_branch.project_id = project_branch_parent.project_id
```

Comment:
Just a question; is the `project_id` check actually required here, or is it just to get an index to kick in or something?

Reply:
It's written this way because a branch's conceptual (and actual) primary key is the (project id, branch id) pair, even though we use UUIDs for branch ids